spring-pulsar/src/main/java/org/springframework/pulsar/listener/reactive/ReactivePulsarMessageHandler.java

```diff
@@ -16,18 +16,11 @@
 
 package org.springframework.pulsar.listener.reactive;
 
-import org.apache.pulsar.client.api.Message;
-import org.reactivestreams.Publisher;
-
 /**
- * Message handler class with a {@link #received} method for use in
- * {@link org.apache.pulsar.reactive.client.api.ReactiveMessagePipelineBuilder#messageHandler}.
+ * Reactive message handler used by {@link DefaultReactivePulsarMessageListenerContainer}.
  *
- * @param <T> message payload type
  * @author Christophe Bornet
  */
-public interface ReactivePulsarMessageHandler<T> {
-
-	Publisher<Void> received(Message<T> message);
+public sealed interface ReactivePulsarMessageHandler permits ReactivePulsarOneByOneMessageHandler, ReactivePulsarStreamingHandler {
```

Comment:
Any reason not to type RPMH to `<T>` as well?

Reply:
I didn't type it bc it doesn't have anything that requires a type.
But maybe we can add it if we have need for types later

Reply:
Got it. Yeh, if we add the type it also allows us to remove the unchecked suppressions in quite a few places in the container impl.

Reply:
If I add it. I get a warning from IDEA that `T` is never used...

Reply:
And it doesn't help to remove any unchecked

Reply:
Haha - can't win for losing w/ generics and warnings 😸